### PR TITLE
UI: Add option to "Hide Transitions widgets" in Studio Mode

### PR DESCRIPTION
**Pull request by kamalmostafa**
### Description

New Settings option: General -> Studio Mode -> Hide Transition widgets

When checked, the Transition button and associated widgets are hidden, allowing the side-by-side Preview and Program video windows to display as large as possible.

Note: When the Transition widgets are hidden, transitions can still be invoked by the "Transition to scene when double-clicked" method, websocket, or scripting methods.

obs-HideTransitionWidgets

### Motivation and Context

https://ideas.obsproject.com/posts/1129/studio-mode-ability-to-hide-transition-panel

Closes obsproject/obs-studio#7592

### How Has This Been Tested?

Functional testing of the option switch behavior on Ubuntu 22.04.

### Types of changes

- Tweak (non-breaking change to improve existing functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [ ] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [ ] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by Fenrirthviti** [comment]
Please correct the PR description to properly fill out the template.

**Comment by kamalmostafa** [comment]
- properly fill out the template
 - fixed clang-format
 - rebased

**Comment by cg2121** [comment]
This should be in the view menu not the settings.

**Comment by kamalmostafa** [comment]
> This should be in the view menu not the settings.

I think it's better in Settings -> General -> Studio Mode, because:

- This is presumably a quite rare thing to want to enable, only suitable for a small number of (advanced) users.  Why clutter the View menu with it?

- It is *only* relevant for Studio Mode, and could easily confuse a (new) user who might select the "hide" option if it were in the View menu before ever switching to Studio Mode for the first time -- there would be no visible effect to it.  But later when they first explore Studio Mode, they'll never even see the Transition widgets at all (and have no idea that they're "missing").

- It makes sense to keep this hide switch *right* next to "Transition to scene when double-clicked" (which is in Settings), since many users will want to ensure that the latter is enabled, if that's how they intend to invoke their transitions.  The Studio Mode block in Settings contains other switches which affect both layout and functionality.

Are there stronger reasons why it should be in the View menu?  (Really, whichever 50 lines of framework invoke the single `setVisibility()` call that this all boils down to is fine with me :smile:).

File: UI/window-basic-main.cpp
Comment (tytan652) [inline UI/window-basic-main.cpp:4378]: Mismatch with others.
```suggestion
		App()->GlobalConfig(), "BasicWindow", "HideTransitionWidget");
```

**Review by tytan652 — CHANGES_REQUESTED** [review]

**Comment by kamalmostafa** [comment]
Huh.  I clicked the "Request re-review" button for @tytan652 ... and that seems to have removed the review request for @Warchamp7 (that was not my intention).

**Review by tytan652 — COMMENTED** [review]
What I asked to fix on my first review was fixed.

**Comment by Fenrirthviti** [comment]
Can we have a clear problem statement on what exact problem this is solving? Is this simply a space issue, or is this allowing a workflow that doesn't otherwise exist? Have you considered using a multi-view projector for a cleaner view, and popping out the required docks for a better/more compact setup with a similar end result?

As it stands, I have one fundamental issue with this solution, which is that when you hide the transition controls (as mentioned) you lose functionality in the program unless you know hidden information, which is a really bad situation to be in. I agree that putting it in the Settings menu means it's less likely that someone will accidentally turn the transition controls off, but this is still a concern.

I just want to make sure we fully understand what the problem is, and that this is the correct solution to that problem as it's a little ambiguous right now.

As an aside, is this something that has always been an issue, or is it only now an issue due to the new default theme?

**Comment by kamalmostafa** [comment]
Thanks for your review @Fenrirthviti, 

> Can we have a clear problem statement on what exact problem this is solving? Is this simply a space issue, or is this allowing a workflow that doesn't otherwise exist?

This is purely a "space issue"; no new workflow is enabled.  The purpose is to make better use of the screen real-estate "wasted" by the Transition widget panel, for setups where those buttons are never used (like mine, where all transition control is by websocket).  I'd much rather have bigger Preview/Program video windows than a bunch of buttons that I never touch.  Presumably, the other commenters in the ideas.obsproject.com topic have similar reasons for wanting to hide them.
 
> Have you considered using a multi-view projector for a cleaner view, and popping out the required docks for a better/more compact setup with a similar end result?

I experimented with this, but it did not provide a better layout for me.  I note that there isn't actually a 2x1 "just-Preview/Program" multi-view panel.  And I don't particularly want the video windows detached from the main window anyway.

On the topic of pop-out docks though ... Another idea mentioned in the ideas.obsproject.com topic was to make the Transition widgets panel a detachable dock, and that would also allow for my desired layout, of course.  (The "Scene Transitions" detachable dock even looks awfully similar to the widget panel.  Hmmm.).

> As it stands, I have one fundamental issue with this solution, which is that when you hide the transition controls (as mentioned) you lose functionality in the program unless you know hidden information, which is a really bad situation to be in. I agree that putting it in the Settings menu means it's less likely that someone will accidentally turn the transition controls off, but this is still a concern.

A fair concern, though I'm not sure this switch is any "more dangerous" than myriad other settings checkboxes.

That said, I've added another commit [8cee669](https://github.com/obsproject/obs-studio/pull/7593/commits/8cee669ac931acb5835af480bf35aecd77e5fb27) which adds an indicator label "Note" whenever the hidden Transisition widgets are being hidden.  I'm not in love with it, but it does address the "hidden information" concern.  Meh?

obs-HiddenTransitionWidgetsLabel

 
> I just want to make sure we fully understand what the problem is, and that this is the correct solution to that problem as it's a little ambiguous right now.
> 
> As an aside, is this something that has always been an issue, or is it only now an issue due to the new default theme?

This is not related to the new theme; I carried a hack version of this in v27 too and the ideas.obsproject.com topic is years old.

**Comment by Fenrirthviti** [comment]
> This is purely a "space issue"; no new workflow is enabled. The purpose is to make better use of the screen real-estate "wasted" by the Transition widget panel, for setups where those buttons are never used (like mine, where all transition control is by websocket). I'd much rather have bigger Preview/Program video windows than a bunch of buttons that I never touch. Presumably, the other commenters in the ideas.obsproject.com topic have similar reasons for wanting to hide them.

Thanks for the confirmation here.

> A fair concern, though I'm not sure this switch is any "more dangerous" than myriad other settings checkboxes.

I don't want sins of the past informing future decisions, though I'd be curious what you're referring to as we're making a concerted effort to reduce the amount of dangerous or hidden options. This is probably better suited off-thread here though, I'd definitely love to hear your thoughts on our Discord, if you ping me (Fenrir) in the #development channel.
 
> That said, I've added another commit [8cee669](https://github.com/obsproject/obs-studio/pull/7593/commits/8cee669ac931acb5835af480bf35aecd77e5fb27) which adds an indicator label "Note" whenever the hidden Transisition widgets are being hidden. I'm not in love with it, but it does address the "hidden information" concern. Meh?
> 
> image snipped

This uses the space reserved for the source control bar, so not a viable solution unfortunately.

In general, I'm unhappy with this as a solution as I feel it's not exactly the correct way to solve the problem. Internally (and in some public places I don't have handy) we've discussed changing the transition settings to a separate dialog where you can configure and set up transitions, set a matrix for to/from scenes for individual transitions, manage source show/hide transitions, etc., and while that hasn't been a high priority for the project as a whole, it's something that might alleviate some of the issues here.

However, the transition controls themselves are something that are fundamental to how Studio Mode works, and information that shouldn't ever be hidden, IMO. This is just my opinion, so don't take this as a rejection or that this won't be merged, I just wanted to voice my concerns. I understand that for advanced users who are absolutely sure of what they are doing, it might make sense as they're moving the functionality somewhere else, but we as maintainers need to ask if this helping or beneficial to a majority of the userbase, or if this cause more harm to the average user who doesn't understand what they're doing. This is basically an impossible question to answer as we have no user telemetry, and it's hard to set aside bias and preconceptions about the program as a whole.

@Warchamp7 will also need to weigh in here as the current UI/UX decision maker before a final decision can be made.

**Comment by Warchamp7** [comment]
Sorry for the long delay on responding to this.

I appreciate the desire to remove unused space in certain workflows, but unfortunately this makes Studio Mode non-functional on a fundamental level without having explicitly enabled other settings (Ex. Double click to switch scenes) or other workflow methods (Hotkeys, third-party devices/plugin API calls) and I am not comfortable with that on a UX level.

**Pull request closed without merging**
